vtex/faststore
# Offline page support for PWAs

## What's the purpose of this pull request?

This PR adds a few necessary changes in order for our stores to function correctly when offline. In `gatsby-theme-store`, there's a default offline page implementation, which can then be replaced by each store.

There's still a bit of configuration that's being done in the store's theme. In the future, we should probably extract all configuration into a new Gatsby plugin.

This change along with https://github.com/vtex-sites/storecomponents.store/pull/701 also guarantees that our stores pass the new installation criteria for PWAs that Chrome is going to use. More info on that here: https://developer.chrome.com/blog/improved-pwa-offline-detection/.

## How it works? 

This PR alone doesn't do much, but it basically prevents our existing error handlers from throwing whenever there's a connection issue. And it adds a new route to every store `/offline`. This is where you can access the default offline page.

## How to test it?

This PR by itself doesn't result in visible changes, but along with https://github.com/vtex-sites/storecomponents.store/pull/701, it can be tested in the following deploy preview: https://sfj-8ffff3b--storecomponents.vtex.app/

1. Open the link above to access storecomponents.store.
2. Wait a few seconds and interact with the store
3. Open Chrome dev tools and set your connection to "offline"
4. Try going into any other store page. If it's a page you haven't already been to, you're going to see the default offline page.

## References
<em>Spread the knowledge: is there any content you used to create this PR that is worth sharing?</em>  

<em>Extra tip: adding references to related issues or mentioning people important to this PR may be good for the documentation and reviewing process</em>
